airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftInsertDestination.java

```diff
@@ -45,12 +45,19 @@ public class RedshiftInsertDestination extends AbstractJdbcDestination<RedshiftS
       JdbcUtils.SSL_KEY, "true",
       "sslfactory", "com.amazon.redshift.ssl.NonValidatingFactory");
 
+  // insert into stmt has ~200 bytes
+  // Per record overhead of ~150 bytes for strings in statement like JSON_PARSE.. uuid etc
+  // If the flush size allows the max batch of 10k records, then net overhead is ~1.5MB.
+  // Lets round it to 2MB and keep a max buffer of 14MB
+  // This will allow not sending record set larger than 14M limiting the batch insert statement.
+  private static final Long MAX_BATCH_SIZE_FOR_FLUSH = 14 * 1024 * 1024L;
```

Comment:
nit: how is the optimalBatchSize parameter used? I.e. does the async framework treat it as a recommendation, or a hard limit? (and let's align the naming to either optimalBatchSize + OPTIMAL_BATCH_SIZE_FOR_FLUSH, or maxBatchSize + MAX_BATCH_SIZE_FOR_FLUSH)